### DISCUSSION TITLE
Drag-and-drop not working for nested array in useFieldArray

### DISCUSSION THREAD

**Discussion by miraj09** [react-hook-form/react-hook-form#9672] [Q&A] [answered]
I am trying to implement the drag-n-drop feature on a nested array of inputs. I am using `react-beautiful-dnd` and `react-hook-form` libraries: `useFieldArray` hook to manage the form and `useController` hook for controlled input.

The problem I am facing is when i try to reorder any child input, change happens in the last item only. here is a [sandbox](https://codesandbox.io/s/watch-usewatch-calc-forked-t5ltk2?file=/src/nestedFieldArray.js).

To reproduce the issue, you need to add at least 2 top-level inputs along with 2 nested inputs for each of those. Next try to reorder the nested inputs of the 1st top-level item. As a result, child inputs of the 2nd item would change their positions, while the initially dragged inputs would revert back to their original position. [![see GIF](https://i.stack.imgur.com/TgX55.gif)](https://i.stack.imgur.com/TgX55.gif)

I tried to pass `ref` to nested inputs and used `useImperativeHandle` hook to expose `move` method of `useFieldArray` hook from child component to reorder. But everytime only the last child gets reordered

**Comment by nimeshmaharjan1** [comment]
use react-beautiful-dnd

**Reply by nimeshmaharjan1** [reply]
> use react-beautiful-dnd

oh sorry I thought you were using react dnd

check your reordering logic when the onDragEnd function is triggered in the DragDropContext

**Reply by nimeshmaharjan1** [reply]
```js
const handleDragEnd = async (result: DropResult) => {
    const { source, destination } = result;
    if (!destination) {
      return;
    }
    const items = reorder(
      useFormMethods.getValues('surveyQuestions') as [],
      source.index,
      destination.index
    );
    useFormMethods.setValue(
      'surveyQuestions',
      items.map((item) => ({
        ...(item as SurveyQuestion),
        survey_question_modal_position: items.indexOf(item),
      }))
    );
  };
```

**Reply by nimeshmaharjan1** [reply]
```js
export const reorder = (list: [], startIndex: number, endIndex: number) => {
  const result = Array.from(list);
  const [removed] = result.splice(startIndex, 1);
  result.splice(endIndex, 0, removed);

  return result;
};
```

**Reply by miraj09** [reply]
@nimeshmaharjan1 
thanks for your response.

**Comment by leapful** [comment, marked as answer]
@miraj09 

You are using same `ref` for your nested field array to store reorder child function therefore when appending new nested child field array, it will override the reorder function and only move item belongs to last nested field array.

Just keep a list of the reorder function for each nested field array to avoid overridden on reordering function.

Working demo: https://codesandbox.io/s/watch-usewatch-calc-forked-5vrcsk?file=/src/fieldArray.js

**Reply by miraj09** [reply]
@leapful **Thanks a lot**
